src/front/glsl_new/ast.rs

```diff
@@ -112,3 +112,16 @@ pub struct VarDeclaration {
     pub ids_initializers: Vec<(String, Option<ExpressionRule>)>,
     pub ty: Handle<Type>,
 }
+
+#[derive(Debug)]
+pub enum FunctionCallKind {
+    TypeConstructor(Handle<Type>),
+    Function(Handle<Expression>),
```

Comment:
NOTE: GLSL spec defines the function identifier in a call as a postfix expression, not sure how this will work with naga, looks like the `Call` expression needs a `FunctionOrigin` directly. Will need to figure out when implementing function calls (for now it only handles type ctors)

Reply:
After some more digging, I think this is done in order to support GLSL subroutines (didn't even know there was such a thing). They work by dispatching the function call based on a uniform input.

I suggest we don't support this for now.